### PR TITLE
chore: Angular 16 support

### LINKED ISSUE #984: Angular 16 support
ui-router seem to work fine with Angular 16. However, the peer dependency against Angular 15 breaks the npm install process. It can be overridden using `--force` or `--legacy-peer-deps`, but that feels too hacky for production. We are in the process of migrating to Angulars built in routing, but are not done yet. Would it be possible to bump the versions?

Thanks!

### PR DESCRIPTION
**Pull request by boxmein**
This PR makes ui-router/angular target Angular 16 in peerDependencies and during development. 

Fixes #984 

- [x] Update dependencies to Angular 16
- [x] Update Jest, jest-preset-angular and test body to match Angular 16
- [x] Bump major version (because of peerDependency changes)
- [x] test-typescript-versions 4.9 generated
- [x] test-angular-versions 16 generated
- [x] These sections are not completed: https://github.com/ui-router/angular/blob/master/add_support_for_new_angular_version.txt#L21-L50

I've verified that the commits build and tests are passing. Where documentation was mentioning entryComponents, it was updated to remove such mentions.

### CODE REVIEW

**Comment by boxmein** [comment]
@wawyed Now I'm stuck because it appears `test:downstream` cannot be completed without updating sample apps to Angular 16. And sample apps can't be updated because an Angular 16 compatible ui-router is not available.

**Comment by wawyed** [comment]
We need to do the examples in order to get the build to pass

**Comment by wawyed** [comment]
> @wawyed Now I'm stuck because it appears `test:downstream` cannot be completed without updating sample apps to Angular 16. And sample apps can't be updated because an Angular 16 compatible ui-router is not available.

@boxmein create a branch with the changes to 16 for the sample angular example and then reference it in the test downstream json file. Don't worry about that as the script will use the local version when testing which is 16 compatible.

File: downstream_projects.json [hunk 1]
```diff
@@ -1,8 +1,10 @@
 {
   "packageDir": "./dist",
   "projects": {
-    "sample-app-angular": "https://github.com/ui-router/sample-app-angular.git",
+    "sample-app-angular": "https://github.com/boxmein/sample-app-angular.git@boxmein_angular16",
     "angular15": "./test-angular-versions/v15",
```
Comment (wawyed) [inline downstream_projects.json:5]: remove angular 15 i'd say

Comment (boxmein) [reply downstream_projects.json:5]: 💯

File: test-angular-versions/v16/package.json [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+    "@angular/compiler-cli": "^16.1.0",
+    "@types/jasmine": "~4.3.0",
+    "jasmine-core": "~4.6.0",
+    "karma": "~6.4.0",
+    "karma-chrome-launcher": "~3.2.0",
+    "karma-coverage": "~2.2.0",
+    "karma-jasmine": "~5.1.0",
+    "karma-jasmine-html-reporter": "~2.1.0",
+    "typescript": "~4.9.5"
+  }
```
Comment (wawyed) [inline test-angular-versions/v16/package.json:41]: add this 
```
  "checkPeerDependencies": {
    "ignore": [
      "ajv"
    ]
  }
  ```

**Comment by boxmein** [comment]
Right now it looks like the v16 run completed thanks to your review. What are the next steps for me?

**Comment by wawyed** [comment]
I just merged the sample-app-angular  can you change the test_downstream.json

**Comment by boxmein** [comment]
Will do, just a moment

**Pull request merged**